File: catfish-ui/src/components/shared/form-models/text.ts
/**
 * Basic model that stores a string in a given language
 * */

import { Guid } from "guid-typescript"
import { TextType } from "./field";

export interface Text {

    /**
     * Unique ID
     * */
    id: Guid;

    /**
     * The text string enclosed in this object
     * */
    value: string | null;

    /**
     * The type of text input taken by this Text object. */
    textType: TextType | null;

    /**
     * Language
     * */
    lang: string | null;
}

File: Catfish/CatfishExtensions/DTO/Forms/Text.cs
using CatfishExtensions.Constants;

namespace CatfishExtensions.DTO.Forms
{
    public class Text
    {
        public Guid Id { get; set; }
        public string? Value { get; set; }
        public eTextType TextType { get; set; }
        public string Lang { get; set; }
    }
}
